cloud_functions/climateiq_spatialize_chunk_predictions_cf/main_test.py

```diff
@@ -1,47 +1,54 @@
-from io import StringIO
+import base64
+import main
+import pytest
 
 from cloudevents import http
+from io import StringIO
 from google.cloud import firestore
 from google.cloud import storage
 import pandas as pd
 from pandas import testing as pd_testing
-import pytest
-import main
 from unittest import mock
 from typing import Any, Dict
 
 
-def test_export_model_predictions_invalid_object_name() -> None:
-    attributes = {
-        "type": "google.cloud.storage.object.v1.finalized",
-        "source": "source",
-    }
-    data = {
-        "bucket": "climateiq-predictions",
-        "name": "invalid_name",  # Invalid object name
-    }
-    event = http.CloudEvent(attributes, data)
+def test_spatialize_chunk_predictions_invalid_object_name() -> None:
+    event = http.CloudEvent(
+        {
+            "type": "google.cloud.pubsub.topic.v1.messagePublished",
+            "source": "source",
+        },
+        {
+            "message": {
+                "data": base64.b64encode(b"invalid_name"),
+            }
+        },
+    )
 
     with pytest.raises(ValueError) as exc_info:
-        main.export_model_predictions(event)
+        main.subscribe(event)
 
     assert "Invalid object name format. Expected 5 components." in str(exc_info.value)
 
 
 @mock.patch.object(storage, "Client", autospec=True)
 @mock.patch.object(firestore, "Client", autospec=True)
-def test_export_model_predictions_missing_study_area(
+def test_spatialize_chunk_predictions_missing_study_area(
     mock_firestore_client, mock_storage_client
 ) -> None:
-    attributes = {
-        "type": "google.cloud.storage.object.v1.finalized",
-        "source": "source",
-    }
-    data = {
-        "bucket": "climateiq-predictions",
-        "name": "prediction-type/model-id/study-area-name/scenario-id/chunk-id",
-    }
-    event = http.CloudEvent(attributes, data)
+    event = http.CloudEvent(
+        {
+            "type": "google.cloud.pubsub.topic.v1.messagePublished",
+            "source": "source",
+        },
+        {
```

Comment:
Nit: Lots of boilerplate. Could simplify to something like `createPubsubEvent(data=<...>)`